dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SelectedShippingMethodTest.php

```diff
@@ -0,0 +1,223 @@
+     */
+    private $customerTokenService;
+
+    /**
+     * @param $response
+     * @return mixed
+     */
+    public function getSelectedShippingMethod($response)
+    {
+        return $response['cart']['shipping_addresses'][0]['selected_shipping_method'];
+    }
+
+    protected function setUp()
+    {
+        $this->objectManager = \Magento\TestFramework\Helper\Bootstrap::getObjectManager();
+
+        $this->filterBuilder = $this->objectManager->create(
+            \Magento\Framework\Api\FilterBuilder::class
+        );
+        $this->sortOrderBuilder = $this->objectManager->create(
+            \Magento\Framework\Api\SortOrderBuilder::class
+        );
+        $this->searchCriteriaBuilder = $this->objectManager->create(
+            \Magento\Framework\Api\SearchCriteriaBuilder::class
+        );
+        $this->quoteResource = $this->objectManager->get(
+            \Magento\Quote\Model\ResourceModel\Quote::class
+        );
+        $this->quoteFactory = $this->objectManager->get(
+            \Magento\Quote\Model\QuoteFactory::class
+        );
+        $this->quoteIdToMaskedId = $this->objectManager->get(
+            \Magento\Quote\Model\QuoteIdToMaskedQuoteIdInterface::class
+        );
+        $this->customerTokenService = $this->objectManager->get(
+            \Magento\Integration\Api\CustomerTokenServiceInterface::class
+        );
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_shipping_method.php
+     */
+    public function testGetCartWithShippingMethod()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId('test_order_1');
+        $query = $this->getQuery($maskedQuoteId);
+        $response = $this->graphQlQuery($query, [], '', $this->getHeaderMap());
+
+        self::assertArrayHasKey('cart', $response);
+        self::assertInternalType(
+            'array',
+            $response['cart']['shipping_addresses'][0]['selected_shipping_method'],
+            'There are no selected shipping method for customer cart!'
+        );
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_shipping_method.php
+     */
+    public function testGetShippingMethodFromCustomerCart()
+    {
+        $maskedQuoteId = $this->getMaskedQuoteIdByReservedOrderId('test_order_1');
+        $query = $this->getQuery($maskedQuoteId);
+        $response = $this->graphQlQuery($query, [], '', $this->getHeaderMap());
+
+        self::assertArrayHasKey('cart', $response);
+
+        $selectedShippingMethod = $this->getSelectedShippingMethod($response);
+
+        self::assertEquals('flatrate', $selectedShippingMethod['carrier_code']);
+        self::assertEquals('flatrate', $selectedShippingMethod['method_code']);
+        self::assertEquals('Flat Rate - Fixed', $selectedShippingMethod['label']);
+        self::assertEquals(0, $selectedShippingMethod['amount']);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_address.php
```

Comment:
I guess there should be annotation like
`* @expectedException \Magento\Framework\Exception\NoSuchEntityException`